### PR TITLE
Add support for different number of spare and parity units in motr

### PR DESCRIPTION
**Pull request by ankit-seagate**
Earlier spare units were calculated based on the number of parity units(K)
defined. It was assumed that spare units are equal to parity units.
To separate the two, I have introduced a new layout parameter 'S'
indicating number of spare units and changed the parity declusturing
formula from N+2*K to N+K+S.

### CODE REVIEW

**Comment by ankit-seagate** [comment]
This is the initial draft, Kindly review. Meanwhile documentation changes are being worked upon.

File: fd/ut/common.h [hunk 1]
```diff
@@ -28,6 +28,7 @@
 enum layout_attr {
 	la_N = 8,
         la_K = 2,
+	la_S = 2
```
Comment (huanghua78) [inline fd/ut/common.h:31]: alignment issue on previous line.

**Review by huanghua78 — APPROVED** [review]

**Comment by madhavemuri** [comment]
@ankit-seagate : Can you check with @mssawant regarding any corresponding changes in HAre ?

**Comment by sergey-shilov** [comment]
@ankit-seagate please consider dix_rop_tgt_iter_max() in dix/req.c.

**Comment by huanghua78** [comment]
Hello @ankit-seagate , what is the status of this PR?
@madhavemuri @mssawant @yatin-mahajan please review this PR.

**Comment by mssawant** [comment]
@ankit-seagate,  Note that number of spare units can be either 0 or K (parity units). It doesn't make sense to support 0 <= S <= K, it must be S = 0 or S = K.
I did not find  0 <= S <= K in your patch but for the sake of QA I think it is good to document somewhere. Also it would be good to assert this condition during Motr initialization.

**Comment by huanghua78** [comment]
@ankit-seagate Usually, we don't use " force-pushed" to update the PR.
Otherwise, the history is wiped out, and we can not tell the changes in new commit.

**Comment by ankit-seagate** [comment]
> @ankit-seagate Usually, we don't use " force-pushed" to update the PR.
> Otherwise, the history is wiped out, and we can not tell the changes in new commit.

Ok. Will keep it in mind

> @ankit-seagate, Note that number of spare units can be either 0 or K (parity units). It doesn't make sense to support 0 <= S <= K, it must be S = 0 or S = K.
> I did not find 0 <= S <= K in your patch but for the sake of QA I think it is good to document somewhere. Also it would be good to assert this condition during Motr initialization.

Will add the assert

**Comment by huanghua78** [comment]
Hello @ankit-seagate , what is the status of this PR?
Do you plan to update the patch, or are you waiting for it to be reviewed?

@madhavemuri @yatin-mahajan , please provide your review.

**Comment by huanghua78** [comment]
Hello @ankit-seagate , what is the status of this PR?
Do you plan to update the patch, or are you waiting for it to be reviewed?

@madhavemuri @yatin-mahajan , please provide your review.

**Comment by huanghua78** [comment]
Hello @ankit-seagate , what is the status of this PR?
Do you plan to update the patch, or are you waiting for it to be reviewed?

@madhavemuri @yatin-mahajan , please provide your review.

**Comment by gowthamchinna** [comment]
test this please

**Comment by huanghua78** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/301/
is triggered.

**Comment by huanghua78** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/305/
is triggered.

**Comment by huanghua78** [comment]
#593 is created for this PR.

**Pull request closed without merging**